src/bin/vip-wp.js

```diff
@@ -72,6 +112,66 @@ commandWrapper( {
 
 		let result;
 		let rl;
+		let subShellRl;
+
+		if ( isSubShell ) {
+			console.log( `Entering WP-CLI subshell mode for ${ formatEnvironment( envName ) } on ${ appName } (${ appId })` );
+
+			subShellRl = readline.createInterface( {
+				input: process.stdin,
+				output: process.stdout,
+				terminal: true,
+				prompt: `${ appName }.${ envName }> `,
+				// TODO make history persistent across sessions for same env
+				historySize: 200,
+			} );
+
+			subShellRl.on( 'line', async line => {
+				console.log( 'received line:', line );
+				subShellRl.pause();
+
```

Comment:
We should do some basic validation here to ensure it looks like a real command, before sending it to server. So no empty lines, starts with `wp `, etc.

Reply:
I added a check for empty line and if it starts with `wp `. Do you think of anything else we can add ? https://github.com/Automattic/vip/pull/280/commits/cc2faa665cb3e1ffb1d3a38473b395616facad6d

Reply:
Awesome, that should be good for now.